sys/xtimer/xtimer_core.c

```diff
@@ -176,42 +176,30 @@ static void _add_timer_to_list(xtimer_t **list_head, xtimer_t *timer)
 /**
  * @brief remove a timer from an ordered list of timers
  */
-static int _remove_timer_from_list(xtimer_t **list_head, xtimer_t *timer)
+static void _remove_timer_from_list(xtimer_t **list_head, xtimer_t *timer)
 {
     while (*list_head) {
         if (*list_head == timer) {
             *list_head = timer->next;
-            timer->offset = 0;
-            timer->long_offset = 0;
-            timer->start_time = 0;
             timer->next = NULL;
-            return 1;
         }
         list_head = &((*list_head)->next);
     }
-
-    return 0;
 }
 
 void xtimer_remove(xtimer_t *timer)
 {
-    bool reschedule = false;
-
     /* time sensitive since the target timer can be fired */
     unsigned int state = irq_disable();
-    if (timer == timer_list_head) {
-        reschedule = true;
-    }
+    timer->offset = 0;
+    timer->long_offset = 0;
+    timer->start_time = 0;
 
-    if (!_remove_timer_from_list(&timer_list_head, timer)) {
+    if (timer->long_offset) {
```

Comment:
the first line in combination with this condition is probably not what you want ;)
[If you received multiple notifications on this, it was just me being too stupid for the multiline comments, sorry for the noise^^]

Reply:
I'm not sure if I fully understand your comment here. If long_offset is on, that means "timer" is in the long timer list. So I think timer should be removed from the long timer list?

Reply:
I agree, but when you always set `timer->long_offset = 0;` it will always assume that the timer is a short timer. I.e. because you set long_offset to 0 before, it will always jump to the else branch. How to use that function for removing a long timer now? I hope that makes it more clear.

Reply:
Understood. Thanks!